app/controllers/discussions_messages_controller.rb

```diff
@@ -11,7 +11,10 @@ def create
   end
 
   def destroy
-    current_message.destroy!
+    authorize_moderator! unless params[:motive] == :self_deleted.to_s
```

Comment:
This works for now, but I think it would be a good idea to reify `allowed_motives_for_deletion` in the future like:

{ initiatior: [:self_deleted], moderator: [:inappropiate_content, ...] }

And then use them both in the view and the model to keep them synced in an unified place.

It's not a big deal, but a moderator could self_delete a message for an user this way.

Reply:
Could be indeed but we're not doing it for now. I do agree that syncing both view and model would be great.